server/punishments.js

```diff
@@ -404,9 +406,9 @@ Punishments.appendSharedIp = function (ip, note) {
 
 Punishments.saveSharedIps = function () {
 	let buf = 'IP\tType\tNote\r\n';
-	Punishments.sharedIps.forEach((/** @type {string} */ note, /** @type {string} */ ip) => {
+	for (const [/** @type {string} */note, /** @type {string} */ip] of Punishments.sharedIps) {
```

Comment:
No, this isn't a real each-function, you have to use `forEach`.

Reply:
You could also like rewrite the current `forEach` overrides as iterators, but that seems beyond you.